src/main.cpp

```diff
@@ -48,6 +53,41 @@ int runMixxx(MixxxApplication* pApp, const CmdlineArgs& args) {
     }
 }
 
+void adjustScaleFactor(CmdlineArgs* pArgs) {
+    if (qEnvironmentVariableIsSet(kScaleFactorEnvVar)) {
+        bool ok;
+        const double f = qgetenv(kScaleFactorEnvVar).toDouble(&ok);
+        if (ok && f > 0) {
+            // The environment variable overrides the preferences option
+            qDebug() << "Using" << kScaleFactorEnvVar << f;
+            pArgs->storeScaleFactor(f);
+            return;
+        }
+    }
+    // We cannot use ConfigObject, because it depends on MixxxApplication
+    // but the scale factor is read during it's constructor.
+    // QHighDpiScaling can not be used afterwards because it is private.
+    auto cfgFile = QFile(QDir(pArgs->getSettingsPath()).filePath(MIXXX_SETTINGS_FILE));
+    if (cfgFile.open(QFile::ReadOnly | QFile::Text)) {
+        QTextStream in(&cfgFile);
+        QString strScaleFactor;
+        QString line = in.readLine();
```

Comment:
Better use `in.readLine().trimmed()` (here an below) in case users edited their config file manually.